sympy/printing/fcode.py

```diff
@@ -55,15 +57,25 @@ class FCodePrinter(CodePrinter):
     printmethod = "_fcode"
     language = "Fortran"
 
+    type_mappings = {
+        Type('intc'): ('integer(c_int)', {'iso_c_binding': 'c_int'}),
+        float32: ('real(4)', None),
+        float64: ('real(8)', None),
+        complex64: ('complex(4)', None),
+        complex128: ('complex(8)', None),
+        Type('integer'): ('integer', None),
+        Type('bool'): ('logical', None)
+    }
+
     _default_settings = {
         'order': None,
         'full_prec': 'auto',
-        'precision': 15,
+        'precision': 17,
```

Comment:
I changed it from 15 to 17 to preserve precision. There are some issues with printing 17 digits, e.g. the user might find ```
In [2]: Float(2.1).evalf(17)
Out[2]: 2.1000000000000001
```
confusing